R/launcher-functions.R

```diff
@@ -0,0 +1,326 @@
+launcherGetJobs <- function(statuses = NULL,
+                            fields = NULL,
+                            includeSessions = FALSE)
+{
+  callLauncherFun("launcher.getJobs",
+                  statuses = statuses,
+                  fields = fields,
+                  includeSessions = includeSessions)
+}
+
+#' Retrieve Job Information
+#'
+#' Retrieve information on a job with id `jobId`.
+#'
+#' @param jobId The id of a launcher job.
+#'
+#' @export
+launcherGetJob <- function(jobId) {
+  callLauncherFun("launcher.getJob",
+                  jobId = jobId)
+}
+
+#' Define a Launcher Configuration
+#'
+#' Define a launcher configuration, suitable for use with the `config` argument
+#' to [launcherSubmitJob()].
+#'
+#' @param name The name of the launcher configuration.
+#' @param value The configuration value. Must either be an integer, float, or string.
+#'
+#' @family job submission
+#' @export
+launcherConfig <- function(name, value = NULL)
+{
+  valueType <- if (is.integer(value))
+    "int"
+  else if (is.double(value))
+    "float"
+  else if (is.character(value))
+    "string"
+
+  value <- format(value)
+
+  callLauncherFun("launcher.newConfig",
+                  name = name,
+                  value = value,
+                  valueType = valueType)
+}
+
+#' Define a Launcher Container
+#'
+#' Define a launcher container, suitable for use with the `container` argument
+#' to [launcherSubmitJob()].
+#'
+#' @param image The container image to use.
+#' @param runAsUserId The user id to run as within the container. Defaults
+#'   to the container-specified user.
+#' @param runAsGroupId The group id to run as within the container. Defaults
+#'   to the container-specified group.
+#'
+#' @family job submission
+#' @export
+launcherContainer <- function(image,
+                              runAsUserId = NULL,
+                              runAsGroupId = NULL)
+{
+  callLauncherFun("launcher.newContainer",
+                  image = image,
+                  runAsUserId = runAsUserId,
+                  runAsGroupId = runAsGroupId)
+}
+
+#' Define a Launcher Placement Constraint
+#'
+#' Define a launcher placement constraint, suitable for use with the
+#' `placementConstraints` argument to [launcherSubmitJob()].
+#'
+#' @param name The name of this placement constraint.
+#' @param value The value of the constraint. A job will only be placed on a
+#'   requested node if the requested placement constraint is present.
+#'
+#' @family job submission
+#' @export
+launcherPlacementConstraint <- function(name,
+                                        value = NULL)
+{
+  callLauncherFun("launcher.newPlacementConstraint",
+                  name = name,
+                  value = value)
+}
+
+#' Define a Launcher Resource Limit
+#'
+#' Define a launcher resource limit, suitable for use with the
+#' `resourceLimits` argument to [launcherSubmitJob()].
+#'
+#' @param type The resource limit type. Must be one of cpuCount, cpuFrequency,
```

Comment:
The resource limit types are plug-in specific, not all will be supported by every plugin (e.g. not all of these are supported by Kubernetes plugin). Maybe a comment about referring to the launcher plug-in documentation?